src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -373,9 +373,81 @@ begin
         ... = 1 : mul_nonsing_inv B h',
 end
 
-lemma nonsing_inv_right_left (B : matrix n n α) (h : B ⬝ A = 1) : A ⬝ B = 1 :=
+lemma nonsing_inv_right_left (h : B ⬝ A = 1) : A ⬝ B = 1 :=
 B.nonsing_inv_left_right A h
 
+/-- `det A` is a unit implies A is invertible -/
+lemma is_unit_det_of_invertible [invertible A] : is_unit A.det :=
+by apply is_unit_det_of_left_inverse A (invertible.inv_of A) (inv_of_mul_self A)
+
+@[simp,norm]
+lemma inv_eq_nonsing_inv_of_invertible [invertible A] : ⅟ A = A⁻¹ :=
+begin
+  have ha := is_unit_det_of_invertible A,
+  have ha':= (is_unit_iff_is_unit_det A).2 ha,
+  have h := inv_of_mul_self A,
+  have h' := nonsing_inv_mul A ha,
+  rw ←h' at h,
+  apply (is_unit.mul_left_inj ha').1 h,
+end
+
+variables {A} {B}
+
+/-- Matrix A is invertible implies `det A` is a unit. -/
+noncomputable
+lemma invertible_of_is_unit_det  (h : is_unit A.det) : invertible A :=
+⟨A⁻¹, nonsing_inv_mul A h, mul_nonsing_inv A h⟩
```

Comment:
I don't think this defintion (which the linter will tell you should be a `def` not a `is a particularly good idea `invertible` is usually reserved for `instance`s, and `is_unit` is for everything else:
```suggestion
/-- Matrix A is invertible implies `det A` is a unit. -/
lemma is_unit_of_is_unit_det  (h : is_unit A.det) : is_unit A :=
sorry
```

Reply:
Hi, Eric. The suggested lemma "is_unit_of_is_unit_det" has already existed in the file. I wrote "invertible_of_is_unit_det" as it will then be used to prove other newly added lemmas such as B is right invertible implies B is the inverse (I didn't see another quick way to establishing this other than adding "invertible_of_is_unit_det"). Perhaps just change the lemmas the linter pointed out to definitions and put comments like "an end-user should usually not use this" beside them if necessary?

Reply:
How about changing "invertible_of_is_unit_det" to def and deleting "invertible_of_left_inverse" and "invertible_of_right_inverse" both together as the two may seem unnecessary (or change them to def as well)?

Reply:
If the "def" `invertible_of_is_unit_det` still seems improper for users to use outside the file, perhaps use `private def`?

Reply:
The whole point of `invertible` is to be a computable version of `is_unit`. This `def` is `noncomputable` which completely defeats that point.